### PR TITLE
Change return type of getEnumTag in value witness table

### PR DESCRIPTION
**Pull request by WANGJIEKE**
Make the `getEnumTag` function in `include/swift/ABI/ValueWitness.def` return an unsigned integer, matching the function's description in the comment.

@hyp If I remember correctly, you mentioned that some other people might also be interested in this PR during office hours. If so, can you help me mention them? Thanks.

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@rjmccall Is this the correct thing to do, or should the comment be updated instead?

**Comment by hyp** [comment]
@rjmccall and I discussed it , this patch should be ok to merge .

**Comment by hyp** [comment]
We had also discussed an additional test we should add to the Swift compiler to strengthen the contract between IRGen and generated headers/ runtime that calls this function using the C function calling convention. The IRGen implementation just emits this as `i32` returning function, but in theory the C calling convention could require `zext` on this return value , depending on the target . Right now this shouldn't apply to any targets that Swift supports, but this could be a problem in the future. We should add a test that verifies that the C calling convention for this matches our expectation. For example,

```
$ cat test.c
unsigned char getEnumTagi8(void *p);
unsigned getEnumTagi32(void *p);

void test(void *p) {
  getEnumTagi8(p);
  getEnumTagi32(p);
}
```

would give the following IR:

```
declare zeroext i8 @getEnumTagi8(i8*) #1

declare i32 @getEnumTagi32(i8*) #1
```

We should check using Filecheck that `getEnumTagi32` doesn't have `zeroext` for its type.

**Comment by WANGJIEKE** [comment]
@hyp Can you help me invoke the test? Thanks.

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

File: test/Interop/SwiftToCxx/core/unsigned-return-type-no-zext.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+unsigned char getEnumTagi8(void *p);
+unsigned getEnumTagi32(void *p);
+
+void test(void *p) {
+  getEnumTagi8(p);
+  getEnumTagi32(p);
+}
+
+// CHECK: declare noundef zeroext i8 @_Z12getEnumTagi8Pv(i8* noundef) #1
+// CHECK: declare noundef i32 @_Z13getEnumTagi32Pv(i8* noundef) #1
```
Comment (hyp) [inline test/Interop/SwiftToCxx/core/unsigned-return-type-no-zext.cpp:14]: Nice 🤩

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Review by hyp — APPROVED** [review]
Great, thanks so much for this!

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Pull request merged**